src/mjolnir/bssbuilder.cc

```diff
@@ -500,24 +607,70 @@ void BssBuilder::Build(const boost::property_tree::ptree& pt, const std::string&
            std::to_string(bss_by_tile.size()) + " local graphs with " + std::to_string(nb_threads) +
            " thread(s)");
 
-  size_t floor = bss_by_tile.size() / threads.size();
-  size_t at_ceiling = bss_by_tile.size() - (threads.size() * floor);
-  bss_by_tile_t::const_iterator tile_start, tile_end = bss_by_tile.begin();
-
-  for (size_t i = 0; i < threads.size(); ++i) {
-    // Figure out how many this thread will work on (either ceiling or floor)
-    size_t tile_count = (i < at_ceiling ? floor + 1 : floor);
-    // Where the range begins
-    tile_start = tile_end;
-    // Where the range ends
-    std::advance(tile_end, tile_count);
-    // Make the thread
-    threads[i].reset(new std::thread(build, std::cref(pt.get_child("mjolnir")), std::ref(lock),
-                                     tile_start, tile_end));
+  std::vector<BSSConnection> all;
+  {
+    size_t floor = bss_by_tile.size() / threads.size();
+    size_t at_ceiling = bss_by_tile.size() - (threads.size() * floor);
+    bss_by_tile_t::const_iterator tile_start, tile_end = bss_by_tile.begin();
+
+    for (size_t i = 0; i < threads.size(); ++i) {
+      // Figure out how many this thread will work on (either ceiling or floor)
+      size_t tile_count = (i < at_ceiling ? floor + 1 : floor);
+      // Where the range begins
+      tile_start = tile_end;
+      // Where the range ends
+      std::advance(tile_end, tile_count);
+      // Make the thread
+      threads[i].reset(new std::thread(project_and_add_bss_nodes, std::cref(pt.get_child("mjolnir")),
+                                       std::ref(lock), tile_start, tile_end, std::ref(all)));
+    }
+
+    for (auto& thread : threads) {
+      thread->join();
+    }
   }
 
-  for (auto& thread : threads) {
-    thread->join();
+  // the collection is sorted so that the search will be much faster later.
+  boost::sort(all);
```

Comment:
why not `std::sort(all.begin(), all.end());`?

Reply:
because it's shorter with boost :)